jax/_src/numpy/lax_numpy.py

```diff
@@ -2121,9 +2122,25 @@ def array(object: Any, dtype: DTypeLike | None = None, copy: bool = True,
   # to be used for type inference below.
   if isinstance(object, (bool, int, float, complex)):
     _ = dtypes.coerce_to_array(object, dtype)
```

Comment:
Btw, this appears to be no-op: `_` is not used below. I wonder what is the intention of calling `coerce_to_array` or is it a bug?

Reply:
The comment above it describes the intent pretty well I think